src/Transports.AspNetCore.SystemTextJson/GraphQLRequestDeserializer.cs

```diff
@@ -0,0 +1,134 @@
+using GraphQL.Server.Common;
+using GraphQL.Server.Transports.AspNetCore.Common;
+using GraphQL.SystemTextJson;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Buffers;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GraphQL.Server.Transports.AspNetCore.SystemTextJson
+{
+    /// <summary>
+    /// Implementation of an <see cref="IGraphQLRequestDeserializer"/> that uses System.Text.Json;
+    /// reading the request body asynchronously and optimally.
+    /// </summary>
+    /// <remarks>
+    /// With thanks to David Fowler (@davidfowl) for his help getting this right.
+    /// </remarks>
+    public class GraphQLRequestDeserializer : IGraphQLRequestDeserializer
+    {
+        private readonly JsonSerializerOptions _serializerOptions = new JsonSerializerOptions();
+
+        public GraphQLRequestDeserializer(Action<JsonSerializerOptions> configure)
+        {
+            // Add converter that deserializes Variables property
+            _serializerOptions.Converters.Add(new ObjectDictionaryConverter());
+
+            configure?.Invoke(_serializerOptions);
+        }
```

Comment:
```c#
 public GraphQLRequestDeserializer(Action<JsonSerializerOptions> configure)
        {
            configure?.Invoke(_serializerOptions);

            // Add converter that deserializes Variables property
            if (!_serializerOptions.Converters.Any(c => c.CanConvert(typeof(Dictionary<string, object>))))
                _serializerOptions.Converters.Add(new ObjectDictionaryConverter());
        }
```

Reply:
I put the configure call after the addition of the serializer, so that if someone really wanted to override with their own they could. I guess there's an argument to go either way. What do you think?

Reply:
I just got out of my head that one can just remove the converter. Leave as is.